codecov.yml

```diff
@@ -3,10 +3,15 @@ coverage:
   precision: 2
 
 codecov:
+  notify:
+    # Notify via a separate pipeline step in CI once all builds have completed. This way, we don't
+    # report coverage after each parallel build completes, which results in spammy and incorrect
+    # email notifications and prematurely marking the build as having failed.
+    # https://github.com/codecov/codecov-action/issues/1436#issuecomment-2614065472
+    manual_trigger: true
```

Comment:
@ascopes I should have read the PR first, but I think you might just want `after_n_builds` for this kind of behavior

Reply:
There was a discussion I had with one of the developers ages ago about this but at the time I was against doing this since it meant the size of the matrix in the actions were tied to the codecov config, which was a little awkward to remember to keep up to date since I change the matrices when new JDKs get released and when new Maven versions are released.

Will take a look into it again when I get time though.

Thanks for the help on this!

Reply:
Ah, perhaps we should consider making this a dynamic field for these instances, but I can understand that not working with your system